src/xlsx/mod.rs

```diff
@@ -2199,6 +2280,685 @@ pub(crate) fn path_to_zip_path<RS: Read + Seek>(zip: &ZipArchive<RS>, path: &str
+    }
+    pub fn definitions(&self) -> &str {
+        self.definitions.as_ref()
+    }
+}
+
+pub struct PivotCacheIter<'a, RS: Read + Seek + 'a> {
+    definitions: HashMap<String, Vec<(Tag, Value)>>,
+    field_names: Vec<String>,
+    reader: XlReader<'a, RS>,
+}
+fn get_pivot_cache_iter<'a, RS: Read + Seek + 'a>(
+    xl: &'a mut crate::Xlsx<RS>,
+    pivot_table: &PivotTableRef,
+) -> Result<PivotCacheIter<'a, RS>, XlsxError> {
+    let definitions = pivot_table.definitions();
+    let records = pivot_table.records();
+
+    let mut fields: Vec<Vec<(Tag, Value)>> = vec![];
+    let mut definition_map = std::collections::HashMap::new();
+    let mut field_names = vec![];
+
+    // Converting into an iterator requires first reading a pivotCacheDefinitions.xml file
+    // to get lookup values used in pivotCacheRecords.xml file.
+    {
+        let mut xml = match xml_reader(&mut xl.zip, definitions) {
+            None => {
+                return Err(XlsxError::FileNotFound(format!(
+                    "File not found: {}",
+                    definitions
+                )))
+            }
+            Some(x) => x?,
+        };
+
+        let mut buf = Vec::with_capacity(64);
+        // building list of field names and definitions from some pivotCacheDefinitions.xml file
+        loop {
+            buf.clear();
+
+            match xml.read_event_into(&mut buf) {
+                Ok(Event::Start(e)) if e.local_name().as_ref() == b"cacheField" => {
+                    for a in e.attributes() {
+                        match a? {
+                            Attribute {
+                                key: QName(b"name"),
+                                value,
+                            } => {
+                                field_names.push(xml.decoder().decode(value.as_ref())?.to_string());
+                                fields.push(vec![]);
+                            }
+                            Attribute {
+                                key: QName(b"formula"),
+                                value: _value,
+                            } => {
+                                field_names.pop();
+                                fields.pop();
+                            }
+                            _ => {
+                                // do nothing
+                            }
+                        }
+                    }
+                }
+                // Exclude grouped fields from results.
+                // This does not represent the underlying data and should be removed.
+                Ok(Event::Start(e)) if e.local_name().as_ref() == b"groupItems" => {
+                    field_names.pop();
+                    fields.pop();
+                }
+                Ok(Event::Start(e)) if item_tag(&e).is_some() => {
+                    if let Some(field) = fields.last_mut() {
+                        field.push((item_tag(&e).unwrap(), item_value(&e)?));
+                    }
+                }
```

Comment:
Use pattern matching, any `is_some()/unwrap()` combo close by should be a candidate for elimination.

```suggestion
                Ok(Event::Start(e)) => {
                    if let Some(tag) = item_tag(&e) {
                        if let Some(field) = fields.last_mut() {
                            field.push((tag, item_value(&e)?));
                        }
                    }
                }
```